### PR TITLE
Group 6 - Problem 1 Linked List Implementation

### PR DESCRIPTION
**Pull request by alothings**
- Christopher Cordero
- Alonso Gutierrez
- Rodrigo Kimura

### CODE REVIEW

**Comment by Jianyi-Ren** [comment]
Good job! In the **init**, do you guys also point the end to the first element? Shouldnt that be the last element?

**Comment by zapplecat** [comment]
Probably would be good have reverted the commits with all the 30+ c9 files. This is a case where using `git add *` is a bad idea since it makes your repo incredibly messy.

File: linked_list/implementation.py [hunk 1]
```diff
@@ -1,40 +1,135 @@
...
-        pass
+        iterNode = self.start
+        count = 0
+        while iterNode is not None:
+            count += 1
+            iterNode = iterNode.next
+        return count
 
+# http://stackoverflow.com/questions/2180578/python-why-cant-i-iterate-over-a-list-is-my-exception-class-borked
     def __iter__(self):
```
Comment (zapplecat) [inline linked_list/implementation.py:37]: Pretty cool that you got the **iter** method working. I even looked at the same stackoverflow question and didn't figure out the appropriate implementation and ended up going with a generator.

**Comment by alothings** [comment]
@Zizibaluba Good call, we will pay more attention to what we are actually staging and pushing on git.

File: linked_list/implementation.py [hunk 2]
```diff
@@ -1,40 +1,135 @@
...
+        return self
+
+    def __next__(self):
+        if self.iterNode == None:
+            raise StopIteration()
+        value = self.iterNode.elem
+        self.iterNode = self.iterNode.next
+        return value
+        
+    def next(self):
```
Comment (zapplecat) [inline linked_list/implementation.py:49]: Is this just to account for python 2.x? Or is there some other reason?

File: linked_list/implementation.py [hunk 3]
```diff
@@ -1,40 +1,135 @@
...
+            print(index)
+
+        if index > self.__len__() -1:     # if index larger than size
+            raise IndexError
+        
+        if self.start is None:      # if empty list
+            raise IndexError
+        
+        if self.start.next is None: # if one element, set start and end to None
+            temp = self.start
```
Comment (zapplecat) [inline linked_list/implementation.py:104]: I would consider using something other than `temp` as the variable name even if it's pretty unambiguous what's going on here.

File: linked_list/implementation.py (hunk 2 above)
Comment (alothings) [inline linked_list/implementation.py:49]: You got it, the only reason we added the next(self) is for python 2.x

**Comment by alothings** [comment]
@Jianyi-Ren Good call, we could have done self.end = Node(elements[-1]), but we just wanted to make it into a node, then the append method moves that pointer accordingly.

**Pull request closed without merging**